tool/lib/webqa-canonical-generator.js

```diff
@@ -144,6 +166,21 @@ class CanonicalGenerator {
         return this.class;
     }
 
+    _getArgTypeCount(qname) {
+        const schema = this.class.queries[qname];
+        const count = {};
+        for (let arg of schema.iterateArguments()) {
+            let typestr = entityTypeToString(schema.getArgType(arg.name));
+            if (!typestr)
+                continue;
+            if (typestr in count)
+                count[typestr] += 1;
+            else
+                count[typestr] = 1;
```

Comment:
You can write it more concisely as `count[typestr] = (count[typestr] || 0) + 1`
(not sure it's worth it)